pkg/utils/log/log.go

```diff
@@ -0,0 +1,118 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package log
+
+import (
+	"flag"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/elastic/cloud-on-k8s/pkg/about"
+	"github.com/elastic/cloud-on-k8s/pkg/dev"
+	"github.com/go-logr/zapr"
+	pflag "github.com/spf13/pflag"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/klog"
+	crlog "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	verbosity       = flag.Int("log-verbosity", 0, "Verbosity level of logs (-2=Error, -1=Warn, 0=Info, >0=Debug)")
+	enableDebugLogs = flag.Bool("enable-debug-logs", false, "Enable debug logs")
+)
+
+// BindFlags attaches logging flags to the given flag set.
+func BindFlags(flags *pflag.FlagSet) {
+	flags.AddGoFlag(flag.Lookup("log-verbosity"))
+	flags.AddGoFlag(flag.Lookup("enable-debug-logs"))
+}
+
+// InitLogger initializes the global logger informed by the values of log-verbosity and enable-debug-logs flags.
+func InitLogger() {
+	setLogger(verbosity, enableDebugLogs)
+}
+
+// ChangeVerbosity replaces the global logger with a new logger set to the sepcified verbosity level.
+// Verbosity levels from 2 are custom levels that increase the verbosity as the value increases.
+// Standard levels are as follows:
+// level | Zap level | name
+// -------------------------
+//  1    | -1        | Debug
+//  0    |  0        | Info
+// -1    |  1        | Warn
+// -2    |  2        | Error
+func ChangeVerbosity(v int) {
+	debugLogs := false
+	setLogger(&v, &debugLogs)
+}
+
+func setLogger(v *int, debug *bool) {
+	level := determineLogLevel(v, debug)
+
+	// if the level is higher than 1 set the klog level to the same level
```

Comment:
This comment is a bit confusing. IIUC the `level` in the comment is the verbosity level but the variable is the equivalent for zap.
Maybe be more explicit in the code: `zapLogLevel := determineZapLogLevel` and writes `if the verbosity level` in the comment?